### ISSUE TITLE
False-positive "[RefersToDaggerCodegen] Don't refer to Dagger's internal or generated code"

### ISSUE THREAD

**Issue by davido** [google/error-prone#1599]
Last attempt to upgrade EP to 2.3.3 in Bazel was reverted due to reporting `RefersToDaggerCodegen` bug pattern on generated code.

Second attempt to upgrade EP to 2.3.4 is here: [1].

I published custom java_tool release built from this PR, but only for Linux platform: [2].

I created a reproducer in `rules_kotlin` project to demonstrate the problem: [3] and opened this issue: [4].

```
  $ bazel build --java_toolchain=@remote_java_tools_linux//:toolchain --host_java_toolchain=@remote_java_tools_linux//:toolchain src/main/kotlin/...
DEBUG: /home/davido/.cache/bazel/_bazel_davido/17e96fe2f67c13d592b106b04a8a5443/external/bazel_toolchains/rules/version_check.bzl:45:14: 
Current running Bazel is not a release version and one was not defined explicitly in rbe_autoconfig target. Falling back to '0.23.0'
INFO: Analyzed 23 targets (38 packages loaded, 1391 targets configured).
INFO: Found 23 targets...
INFO: From Building external/com_google_protobuf/libprotobuf_java.jar (123 source files, 1 source jar):
warning: -parameters is not supported for target value 1.7. Use 1.8 or later.
INFO: From Building external/com_google_protobuf/libprotobuf_java.jar (123 source files, 1 source jar) [for host]:
warning: -parameters is not supported for target value 1.7. Use 1.8 or later.
ERROR: /home/davido/projects/rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/BUILD:16:13: Building src/main/kotlin/io/bazel/kotlin/builder/libbuilder.jar (2 source files) and running annotation processors (ComponentProcessor) failed (Exit 1): java failed: error executing command 
  (cd /home/davido/.cache/bazel/_bazel_davido/17e96fe2f67c13d592b106b04a8a5443/execroot/io_bazel_rules_kotlin && \
  exec env - \
    LC_CTYPE=en_US.UTF-8 \
    PATH=/home/davido/.yarn/bin:/home/davido/.config/yarn/global/node_modules/.bin:/home/davido/pgm/apache-maven-3.5.2/bin:/home/davido/bin:/home/davido/.yarn/bin:/home/davido/.config/yarn/global/node_modules/.bin:/home/davido/pgm/google-cloud-sdk/bin:/home/davido/pgm/apache-maven-3.5.2/bin:/home/davido/bin:/home/davido/bin:/usr/local/bin:/usr/bin:/bin:/usr/lib/mit/bin:/usr/lib/mit/sbin \
  external/remotejdk11_linux/bin/java -XX:+UseParallelOldGC -XX:-CompactStrings '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED' '--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED' '--patch-module=java.compiler=external/remote_java_tools_linux/java_tools/java_compiler.jar' '--patch-module=jdk.compiler=external/remote_java_tools_linux/java_tools/jdk_compiler.jar' '--add-opens=java.base/java.nio=ALL-UNNAMED' '--add-opens=java.base/java.lang=ALL-UNNAMED' -jar external/remote_java_tools_linux/java_tools/JavaBuilder_deploy.jar @bazel-out/host/bin/src/main/kotlin/io/bazel/kotlin/builder/libbuilder.jar-0.params @bazel-out/host/bin/src/main/kotlin/io/bazel/kotlin/builder/libbuilder.jar-1.params)
Execution platform: @local_config_platform//:host
bazel-out/host/bin/src/main/kotlin/io/bazel/kotlin/builder/_javac/builder/libbuilder_sourcegenfiles/io/bazel/kotlin/builder/KotlinBuilderComponent_Module_ProvidePluginArgEncoderFactory.java:27: error: [RefersToDaggerCodegen] Don't refer to Dagger's internal or generated code
    return providePluginArgEncoder(module, toolchainProvider.get());
                                  ^
    (see https://errorprone.info/bugpattern/RefersToDaggerCodegen)
bazel-out/host/bin/src/main/kotlin/io/bazel/kotlin/builder/_javac/builder/libbuilder_sourcegenfiles/io/bazel/kotlin/builder/KotlinBuilderComponent_Module_ProvidePluginArgEncoderFactory.java:37: error: [RefersToDaggerCodegen] Don't refer to Dagger's internal or generated code
    return Preconditions.checkNotNull(instance.providePluginArgEncoder(toolchain), "Cannot return null from a non-@Nullable @Provides method");
                                     ^
    (see https://errorprone.info/bugpattern/RefersToDaggerCodegen)
INFO: Elapsed time: 85.065s, Critical Path: 74.15s
INFO: 37 processes: 24 linux-sandbox, 13 worker.
FAILED: Build did NOT complete successfully
```

The file content `KotlinBuilderComponent_Module_ProvidePluginArgEncoderFactory.java` is here: [5].

//CC @cushon @ronshapiro

[1] https://github.com/bazelbuild/bazel/pull/11271
[2] https://github.com/davido/java_tools/releases/tag/javac11-v11.0
[3] https://github.com/bazelbuild/rules_kotlin/pull/320
[4] https://github.com/bazelbuild/rules_kotlin/issues/321
[5] http://paste.openstack.org/show/792978

**Comment by davido** [comment]
The breakage was justified. "@Generated" annotation was missing due to missing dependency on 
javax.annotation, that is still needed when Java 8 is used to emit "@Generated" in Dagger. After adding the missing dependency all is fine again:

```java
@Generated(
    value = "dagger.internal.codegen.ComponentProcessor",
    comments = "https://dagger.dev"
)
@SuppressWarnings({
    "unchecked",
    "rawtypes"
})
```
